src/Tasks/Common/Resources/Strings.resx

```diff
@@ -972,5 +972,9 @@ You may need to build the project on another operating system or architecture, o
     <value>NETSDK1220: UseUwp and all associated functionality require using a TFM of 'net8.0-windows' or greater.</value>
     <comment>{StrBegin="NETSDK1220: "}</comment>
   </data>
-  <!-- The latest message added is WindowsSDKXamlInvalidTfm. Please update this value with each PR to catch parallel PRs both adding a new message -->
+  <data name="MicrosoftNetSdkCompilersToolsetRootEmpty" xml:space="preserve">
+    <value>NETSDK1221: NuGetPackageRoot property is empty so package Microsoft.Net.Sdk.Compilers.Toolset cannot be used but it is recommended because your MSBuild and SDK versions are mismatched. Ensure you are building with '/restore /t:Build' and not '/t:Restore;Build'.</value>
```

Comment:
`/t:Restore;Build` is bad but it's very non-obvious that it is bad.  Do we have documentation covering this that we could also refer to from this error message? @baronfel @marcpopMSFT @nkolev92 @zivkan

Reply:
That is a great question. Overall the build check rules are going to start defining a lot of _bad_ msbuild practices. Guessing we will need a bit of a FAQ page to cover them all. I would expect this discussion to be a good fit for that type of page.

Reply:
https://learn.microsoft.com/nuget/reference/msbuild-targets#restoring-and-building-with-one-msbuild-command

although one day we really, really need to split restore and pack into different docs pages, so I don't expect this URL to be long term permanent.

Reply:
We've got a tracking issue for a BuildCheck for this that goes into _why_ it's so subtle and painfully wrong: https://github.com/dotnet/msbuild/issues/9553

Reply:
Do we have a FAQ style landing page for build check? Basically a page which is here are all the build check warnings that come in the box, their error codes and why we have them?

Reply:
Yeah - @JanKrivanek showed it in Showcase earlier today but all built-in BuildChecks create a link that links to their docs on [this page](https://github.com/dotnet/msbuild/blob/main/documentation/specs/BuildCheck/Codes.md) (though it's all aka.ms links so we can change it to more detailed docs as we require them).